### ISSUE TITLE
len_without_is_empty : warning: struct `Lru` has a public `len` method, but the `is_empty` method has an unexpected signature

### ISSUE THREAD

**Issue by rohitjoshi** [rust-lang/rust-clippy#6972]
Lint name: len_without_is_empty

I tried this code:

```rust
 #[inline(always)]
    pub async fn is_empty(&self) -> bool {
        self.len().await > 0
    }

    #[inline(always)]
    pub async fn len(&self) -> usize {
        self.cache.lock().await.len()
    }
```

I expected to see this happen: *It should not generate warning*

Instead, this happened: *
warning: struct `Lru` has a public `len` method, but the `is_empty` method has an unexpected signature
   --> xxx-async/src/cache/lru_cache.rs:175:5
    |
175 |     pub async fn len(&self) -> usize {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: `#[warn(clippy::len_without_is_empty)]` on by default
note: `is_empty` defined here
   --> xxx-async/src/cache/lru_cache.rs:170:5
    |
170 |     pub async fn is_empty(&self) -> bool {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    = note: expected signature: `(&self) -> bool`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#len_without_is_empty
*

### Meta

- `cargo clippy -V`: clippy 0.1.52 (07e0e2ec 2021-03-24))
- `rustc -Vv`:
  ```
 rustc 1.53.0-nightly (07e0e2ec2 2021-03-24)
binary: rustc
commit-hash: 07e0e2ec268c140e607e1ac7f49f145612d0f597
commit-date: 2021-03-24
host: x86_64-apple-darwin
release: 1.53.0-nightly
LLVM version: 12.0.0
  ```
